collect_app/src/main/java/org/odk/collect/android/widgets/BearingWidget.java

```diff
@@ -147,9 +180,80 @@ private void checkForRequiredSensors() {
             isMagneticFieldSensorAvailable = true;
         }
 
-        if (!isAccelerometerSensorAvailable || ! isMagneticFieldSensorAvailable) {
-            getBearingButton.setEnabled(false);
-            ToastUtils.showLongToast(R.string.bearing_lack_of_sensors);
+        if (!isAccelerometerSensorAvailable || !isMagneticFieldSensorAvailable) {
+            return false;
         }
+
+        return true;
+    }
+
+    private EditText getEditText() {
+        final EditText manualData = new EditText(getContext());
+        manualData.setPadding(20, 20, 20, 20);
+        manualData.setTextSize(TypedValue.COMPLEX_UNIT_DIP, getAnswerFontSize());
+        manualData.setInputType(InputType.TYPE_NUMBER_FLAG_DECIMAL);
+        manualData.setKeyListener(new DigitsKeyListener(true, true));
+        manualData.setFocusable(true);
+        manualData.setFocusableInTouchMode(true);
+
+        TableLayout.LayoutParams params = new TableLayout.LayoutParams();
+        params.setMargins(7, 5, 7, 5);
+        final InputFilter[] fa = new InputFilter[1];
+        fa[0] = new InputFilter.LengthFilter(7);
+        manualData.setFilters(fa);
+        manualData.setLayoutParams(params);
+
+        manualData.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                String data = manualData.getText().toString();
+                if (!data.equals("") && isValidInput(data)) {
+                    answerDisplay.setText(manualData.getText().toString());
+                    ToastUtils.showShortToast("Data entered : " + data);
```

Comment:
I would rather not have these toasts! I think they are confusing. They also don't actually prevent users from entering invalid values. I don't think there should be any validation -- people who need some can add it in their forms instead.

Reply:
Got it. Will remove every type of validation.